tests/integration/integration-test.py

```diff
@@ -48,10 +48,10 @@ def test_get_notes_path(self, endpoint='/notes'):
             # Validating the note received belong to the student id
             # fetching the first 9 digits of the note id
             # fails if it does not have 9 digit id
-            match_student_id = re.match(r'(?<!\d)\d{9}(?!\d)', note_id)
+            match_student_id = re.match(r'(^\d{9})(\b-)', note_id)
```

Comment:
What does `(\b)` mean? is it necessary? Also, for your current regex, this example: `123456789-` will still match, but it's not a valid id.